fedimint-core/src/backup.rs

```diff
@@ -0,0 +1,29 @@
+use std::time::SystemTime;
+
+use fedimint_core::encoding::{Decodable, Encodable};
+use fedimint_core::{impl_db_lookup, impl_db_record};
+use serde::{Deserialize, Serialize};
+
+use crate::db::DbKeyPrefix;
+
+/// Key used to store user's ecash backups
+#[derive(Debug, Clone, Copy, Encodable, Decodable, Serialize)]
+pub struct ClientBackupKey(pub secp256k1_zkp::XOnlyPublicKey);
+
+#[derive(Debug, Encodable, Decodable)]
+pub struct ClientBackupKeyPrefix;
+
+impl_db_record!(
+    key = ClientBackupKey,
+    value = ClientBackupSnapshot,
+    db_prefix = DbKeyPrefix::ClientBackup,
+);
+impl_db_lookup!(key = ClientBackupKey, query_prefix = ClientBackupKeyPrefix);
```

Comment:
Doesn't that belong in `fedimint-client`?

Reply:
This is a server side backup. It kind of belongs to `fedimint-server`, but there was some reason why it had to be here.